ui/src/components/AuthDialog.vue

```diff
@@ -120,19 +131,65 @@ export default {
                 config: JSON.stringify(this.configData),
             });
         },
-        saveCredentialsAndRun() {
-            this.saveCredentials();
-            this.runConfig();
+        async saveCredentialsAndRun() {
+            const doRun = await this.saveCredentials();
+            if (doRun) {
+                this.runConfig();
+            }
         },
-        saveCredentials() {
+        async saveCredentials() {
             if (!this.validateForm()) {
-                return;
+                return false;
             }
             const oracleDsn = this.configData.INPUT.values.ORACLE_DSN;
             const postgresDsn = this.configData.OUTPUT.values.PG_DSN;
+            const testCreds = await this.testCredentials(oracleDsn, postgresDsn);
+            if (oracleDsn.include && testCreds.oracle !== 'OK' ||
+                postgresDsn.include && testCreds.postgres !== 'OK') {
+                    console.log(testCreds.oracle !== 'OK')
+                if (testCreds.oracle !== 'OK') {
+                    this.oracleCredsError = testCreds.oracle;
+                }
+                if (testCreds.postgres !== 'OK') {
+                    this.postgresCredsError = testCreds.postgres;
+                }
+                return false;
+            } 
             sessionStorage.setItem(oracleDsn.value, JSON.stringify({user: this.oracleUser, pass: this.oraclePwd}));
-            sessionStorage.setItem(postgresDsn.value, JSON.stringify({user: this.pgUser, pass: this.pgPwd}));
+            if (postgresDsn.include) {
```

Comment:
this is a fix for something we noticed earlier while testing; it still saves the postgres creds to session storage even if disabled